### PR TITLE
Replace text input react component with web component

### LINKED ISSUE #1805: Deprecation Phase 3  Textinput REACT Component
## Description
We will deprecate Textinput React component.

We need to move the Textinput React component into [Phase 3](https://vfs.atlassian.net/wiki/spaces/DST/pages/2243854357/Plan+-+Deprecation+plan+for+React+components#General-deprecation-strategy). This means that the following steps will need to happen:

- Begin migrating any remaining instances of the React component ourselves. 
- The pace of this work will depend on the team’s availability and the priority of this work over other planned work and obligations.
- Once there are no more uses, remove the React component from the component-library repository

## Tasks
- [ ] Check if the component is still being used in vets-website
- [ ] If it is...
  - [ ] Reach out to teams to check if they have an update pending.
  - [ ] Migrate any remaining instances of the React component ourselves.
  - [ ] Once there are no more uses, remove the React component from the component-library repo:
    - [react-components](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/react-components) package
    - [design system dashboard cli](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/design-system-dashboard-cli) package
    - [storybook](https://github.com/department-of-veterans-affairs/) package
  - [ ] Remove any ESLint rules from [vets-website](https://github.com/department-of-veterans-affairs/vets-website) and the [eslint-plugin](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/master/packages/eslint-plugin).
  - [ ] Remove it from the [vets-website GA script](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/site-wide/component-library-analytics-setup.js#L67-L73).
  - [ ] Remove the related web component(s) from the [web component migrations list](https://design.va.gov/about/developers/using-web-components#web-component-migrations)
- [ ] If it is not...
  - [ ] Remove the React component from the component-library repo:
    - [react-components](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/react-components) package
    - [design system dashboard cli](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/design-system-dashboard-cli) package
    -  [storybook](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/storybook) package.
  - [ ] Remove any ESLint rules from [vets-website](https://github.com/department-of-veterans-affairs/vets-website) and the [eslint-plugin](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/master/packages/eslint-plugin).
  - [ ] Remove it from the [vets-website GA script](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/site-wide/component-library-analytics-setup.js#L67-L73).
  - [ ] Remove the related web component(s) from the [web component migrations list](https://design.va.gov/about/developers/using-web-components#web-component-migrations)

## Acceptance Criteria
- [ ] Deprecated component has been removed entirely.

### PR DESCRIPTION
**Pull request by ataker**
Add functionality to check if field is "dirty" before running validation

## Summary

- Update React TextInput component to web component
- Added "dirty" functionality to only validate after field is touched that was included in the React component

## Related issue(s)

- Closes [Deprecation Phase 3 Textinput REACT Component #1805](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1805)

## Testing done

- Ensured behavior remained the same after change
- Tested locally in Chrome and Firefox

### Default untouched
Screenshot 2023-07-21 at 12 13 17 PM

### Error
Screenshot 2023-07-21 at 12 13 26 PM

### Ready to submit
Screenshot 2023-07-21 at 12 13 31 PM

### Request payload
Screenshot 2023-07-21 at 12 13 07 PM

## What areas of the site does it impact?

Veteran request ID card by email form

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/veteran-id-card/actions/index.js [hunk 1]
```diff
@@ -42,7 +43,13 @@ export function timeoutRedirect() {
 export function setEmail(email) {
   return {
     type: VIC_SET_EMAIL,
-    email,
+    email: email.target.value,
```
Comment (ataker) [inline src/applications/veteran-id-card/actions/index.js:46]: Web component returns email value under `email.target.value`

File: src/applications/veteran-id-card/actions/index.js [hunk 2]
```diff
@@ -42,7 +43,13 @@ export function timeoutRedirect() {
...
+  };
+}
+
+export function setDirty() {
+  return {
```
Comment (ataker) [inline src/applications/veteran-id-card/actions/index.js:51]: Add new action to update dirty status

File: src/applications/veteran-id-card/containers/EmailCapture.jsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { connect } from 'react-redux';
+// eslint-disable-next-line deprecate/import
```
[the remark this answers is not in the document]
Comment (ataker) [reply src/applications/veteran-id-card/containers/EmailCapture.jsx:3]: We'll be back for you later

File: src/applications/veteran-id-card/containers/EmailCapture.jsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ class EmailCapture extends React.Component {
 
   handleSubmit(e) {
     e.preventDefault();
-    this.props.submitEmail(this.props.email.value);
+    this.props.submitEmail(this.props.email);
```
Comment (ataker) [inline src/applications/veteran-id-card/containers/EmailCapture.jsx:16]: Email was previously an object, now is a string. Do not need to access `.value` anymore

File: src/applications/veteran-id-card/containers/EmailCapture.jsx [hunk 5]
```diff
@@ -53,18 +53,19 @@ class EmailCapture extends React.Component {
...
-              errorMessage={this.props.errors && this.props.errors[0].title}
-              label={<span>Email address</span>}
+            <va-text-input
+              error={this.props.errors && this.props.errors[0].title}
+              label="Email address"
               name="email"
               field={this.props.email}
-              onValueChange={update => this.props.setEmail(update)}
+              onInput={update => this.props.setEmail(update)}
+              onBlur={update => this.props.setDirty(update)}
```
Comment (ataker) [inline src/applications/veteran-id-card/containers/EmailCapture.jsx:62]: `onBlur` added to replicate `dirty` functionality

**Review by powellkerry — APPROVED** [review]

**Pull request merged**